Sources/NIOHTTP2/HTTP2FrameParser.swift

```
@@ -0,0 +1,738 @@
+    /// Returned data blobs would include anything of potentially flexible
+    /// length, such as DATA payloads, header fragments in HEADERS or PUSH_PROMISE
+    /// frames, and so on. This is to avoid manually copying chunks of data which
+    /// we could just enqueue separately in sequence on the channel. Generally, if
+    /// we have a byte buffer somewhere, we will return that separately rather than
+    /// copy it into another buffer, with the corresponding allocation overhead.
+    ///
+    /// - Parameters:
+    ///   - frame: The frame to encode.
+    ///   - buf: Destination buffer for the encoded frame.
+    /// - Returns: An array containing zero or more additional buffers to send, in
+    ///            order. These may contain data frames' payload bytes, encoded
+    ///            header fragments, etc.
+    /// - Throws: Errors returned from HPACK encoder.
+    func encode(frame: HTTP2Frame, to buf: inout ByteBuffer) throws -> [IOData] {
+        // note our starting point
+        let start = buf.writerIndex
+        
+//      +-----------------------------------------------+
+//      |                 Length (24)                   |
+//      +---------------+---------------+---------------+
+//      |   Type (8)    |   Flags (8)   |
+//      +-+-------------+---------------+-------------------------------+
+//      |R|                 Stream Identifier (31)                      |
+//      +=+=============================================================+
+//      |                   Frame Payload (0...)                      ...
+//      +---------------------------------------------------------------+
+        
+        // skip 24-bit length for now, we'll fill that in later
+        buf.moveWriterIndex(forwardBy: 3)
+        
+        // 8-bit type
+        buf.write(integer: frame.payload.code)
+        // 8-bit flags -- we don't use padding when encoding in NIO, though
+        buf.write(integer: frame.flags.subtracting(.padded).rawValue)
```

If we don't support padding (which is reasonable at this time), we should throw rather than silently suppress it.